src/base/sexpr.zig

```diff
@@ -0,0 +1,218 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const testing = std.testing;
+
+/// Represents a node in an S-expression tree.
+pub const Node = union(enum) {
+    node: struct {
+        value: []const u8,
+        children: std.ArrayListUnmanaged(Node),
+    },
+    string: []const u8,
+    signed_int: i128,
+    unsigned_int: u128,
+    float: f64,
+
+    /// Create a new node with the given value.
+    /// The value will be duplicated so that it is owned by the node
+    /// and will be freed when the node is destroyed.
+    pub fn newNode(gpa: Allocator, value: []const u8) !Node {
+        const owned_value = try gpa.dupe(u8, value);
+        return .{
+            .node = .{
+                .value = owned_value,
+                .children = try std.ArrayListUnmanaged(Node).initCapacity(gpa, 4),
+            },
+        };
+    }
+
+    // Helper function to append a child node to a parent node
+    fn appendChildUnsafe(self: *Node, gpa: Allocator, child: Node) !void {
```

Comment:
Not for this PR, but I wonder if we can make this safe in general.

Reply:
Yeah, I'm not sure...